tools/lisa-build-asset

```diff
@@ -57,25 +73,108 @@ def list_assets(recipe_dir):
         if script.endswith('.recipe')
     )
 
+def get_recipe_name(asset):
+    return f'{asset}.recipe'
+
 def get_asset_recipe(asset, recipe_dir):
-    return os.path.join(recipe_dir, '{}.recipe'.format(asset))
+    return os.path.join(recipe_dir, get_recipe_name(asset))
+
+def get_asset_property(asset, arch, property, **kwargs):
+    # Read asset's recipe to get a property value.
+    # Raise a ValueError if property is not set.
+    action = f"if [ -z ${{{property}+x}} ]; then exit 1; else printf '%s\\n' ${{{property}[@]}}; fi"
+    try:
+        val = make(asset, arch, [action], **kwargs, get_output=True)
+    except subprocess.CalledProcessError:
+        raise ValueError(f'The property "{property}" if not set.')
+    return val[action].splitlines()
+
+@contextmanager
+def make_chroot(asset, arch, build_dir=None, recipe_dir=None, arch_chroot_dir=None):
+    # Create a chroot at arch_chroot_dir as a contextmanager.
+    recipe = get_asset_recipe(asset, recipe_dir)
+    env = get_env(
+        asset=asset,
+        arch=arch,
+        recipe=recipe,
+        build_dir=build_dir
+    )
+
+    # List the required Alpine packages.
+    alpine_dependencies = get_asset_property(
+        asset, arch, "ALPINE_BUILD_DEPENDENCIES",
+        build_dir=build_dir,
+        recipe_dir=recipe_dir,
+        arch_chroot_dir=arch_chroot_dir
+    )
+    alpine_dependencies_args = ('-p', ' '.join(map(quote, alpine_dependencies))) if alpine_dependencies else None
+
+    # List the required env variables.
+    used_env_variables = get_used_env_var(get_asset_recipe(asset, recipe_dir), env)
+    used_env_variables_args = ('-k', ' '.join(used_env_variables)) if used_env_variables else None
+
+    # Install the chroot.
+    subprocess.check_call([
+        'sudo', os.path.join(get_alpine_dir(), 'alpine-chroot-install'),
+        '-d', arch_chroot_dir,
+        '-i', build_dir,
+        '-a', QEMU_ARCH_NAMES.get(arch, arch),
+        *alpine_dependencies_args,
+        *used_env_variables_args],
+        cwd=build_dir
+    )
+
+    # Manually bind additionnal the recipe directory to the chroot.
+    # We are joining two abspaths, so the second needs to be stripped.
+    arch_chroot_recipe_dir = os.path.join(arch_chroot_dir, recipe_dir.strip(os.path.sep))
```

Comment:
`os.path.realpath` should be used instead of direct manipulation like `.strp(os.path.sep)`, it will provide a much more robust result: the path is normalized and is made absolute